odoo/addons/base/models/res_partner.py

```diff
@@ -606,6 +625,51 @@ def _update_fields_values(self, fields):
                 values[fname] = self[fname]
         return values
 
+    @api.model
+    def _get_eu_prefixed_countries(self):
+        return (
+            self.env.ref('base.europe').country_ids
+            + self.env.ref('base.ch')
+            + self.env.ref('base.no')
+            + self.env.ref('base.uk')
+            + self.env.ref('base.sm')
+        )
+
+    def _check_vat(self, validation="error"):
+        for partner in self:
+            vat, _country_code = self._run_vat_checks(partner.commercial_partner_id.country_id, partner.vat,
+                                               partner_name=partner.name, validation=validation)
+            if vat != partner.vat:  # To avoid unnecessary queries (perf tested)
+                partner.vat = vat
+
+    @api.model
+    def _run_vat_checks(self, country, vat, partner_name='', validation='error'):
+        """ Checks a VAT number syntactically to ensure its validity upon saving.
+
+        :param country: a country to check for
+        :param vat: a string with the VAT number to check.
+        :param partner_name: to put into the error message
+        :param validation: if False, it will only return the formatted vat without checking if it valid.
+            if 'error', an incorrect number will raise and if 'setnull' it will just return an empty vat
+
+        :return: The vat number
+                The country code of the country the VAT number
+                 was validated for, if it was validated. False if it could not be validated
+                 against the provided or guessed country.
+        """
+        assert validation in (False, 'error', 'setnull')
+        return vat, country and country.code or ''
+
+    def _local_fp_exists_for_extra_eu(self):
+        """ Hook used for the _get_required_valid implementation (as base_vat does not depend on account anymore)"""
```

Comment:
How do you feel about reading that "anymore" in 2 years? Wouldn't it look silly?